types/got/index.d.ts

```diff
@@ -140,12 +140,18 @@ declare namespace got {
      * @template Body Response body type.
      */
     interface Hooks<Options, Body extends Buffer | string | object> {
+        init?: Array<InitHook<Options>>;
         beforeRequest?: Array<BeforeRequestHook<Options>>;
         beforeRedirect?: Array<BeforeRedirectHook<Options>>;
         beforeRetry?: Array<BeforeRetryHook<Options>>;
         afterResponse?: Array<AfterResponseHook<Options, Body>>;
     }
 
+    /**
+     * @param options Unnormalized request options.
+     */
+    type InitHook<Options> = (options: Options) => any;
```

Comment:
Documentation [says](https://github.com/sindresorhus/got/blob/master/readme.md#hooksinit) 

> Note: This hook must be synchronous!

Maybe  more appropriate return type here is `void`. `any` type is used for other hooks, because they can return `Promise`.